pkg/scheduling/volumeusage.go

```diff
@@ -130,22 +127,32 @@ func resolveDriver(ctx context.Context, kubeClient client.Client, volumeName str
 			return driverName, nil
 		}
 	}
-	if storageClassName != "" {
-		driverName, err := driverFromSC(ctx, kubeClient, storageClassName)
-		if err != nil {
-			return "", err
-		}
-		if driverName != "" {
-			return driverName, nil
+
+	// This state shouldn't be possible during normal operations but if a PV is deleted and its finalizer is removed it is possible.
+	if storageClassName == "" {
+		logging.FromContext(ctx).Errorf("failed to update volume limits for volume with unbound PVC, no storage class specified")
+		return "", nil
+	}
+	driverName, err := driverFromSC(ctx, kubeClient, storageClassName)
+	if err != nil {
+		// There are two scenarios where a StorageClass may be defined but not found:
+		//  1. The StorageClass was manually deleted and the finalizer removed
+		//  2. The StorageClass never existed and was used to bind the PVC to an existing PV, but that PV was removed
+		if errors.IsNotFound(err) {
+			logging.FromContext(ctx).Errorf("failed to update volume limits for volume with unbound PVC, %w", storageClassName)
+			return "", nil
 		}
+		return "", err
 	}
-	// Driver name wasn't able to resolve for this volume. In this case, we just ignore the
-	// volume and move on to the other volumes that the pod has
-	return "", nil
+	return driverName, nil
 }
 
 // driverFromSC resolves the storage driver name by getting the Provisioner name from the StorageClass
 func driverFromSC(ctx context.Context, kubeClient client.Client, storageClassName string) (string, error) {
+	if storageClassName == "" {
```

Comment:
Given our current codepaths, is this ever actually possible?

Reply:
I don't believe it is anymore, I'll verify and remove it 👍